frontend/src/lib/api.ts

```diff
@@ -153,6 +153,7 @@ export interface AnonymousUser {
     is_anonymous: true
     id: number
     is_online: boolean
+    username: string
```

Comment:
This can be null, so we should type it that way.

However, I think reducing the number of states username can be in would be good. Could we make it so username is "anon" (or _always_ populated with a generated name) such that we can get rid of userHasName checks?

Reply:
I actually agree. The issue is that I don't want to burn a bunch of potentially cool names on the thousands(?) of never-to-be-used-again anonymous profiles. But we can set all existing profiles to a default name ("anon") and then, when the site is used by an anon profile, if the profile's username is the default one, we generate one instead?

Reply:
after discussion in discord, this will always be non-null now. userHasName can go away still

Reply:
Gone away.